kafka-rest/src/main/java/io/confluent/kafkarest/resources/v2/PartitionsResource.java

```diff
@@ -200,13 +216,124 @@ public void onCompletion(
     );
   }
 
-  private boolean topicExists(final String topic)  throws Exception {
-    return ctx.getAdminClientWrapper().topicExists(topic);
+  /**
+   * Get offsets for the given {@code topic} and {@code partition}.
+   *
+   * <p>If {@code timestamp} is not specified, returns a summary with beginning and end offsets.
+   * Otherwise, returns the earliest offset whose timestamp is after the given {@code timestamp} and
+   * its associated timestamp, or empty if no such offset exists.</p>
+   *
+   * @throws io.confluent.rest.exceptions.RestNotFoundException if either {@code topic} or {@code
+   *                                                            partition} don't exist.
+   */
+  @GET
+  @Path("/{partition}/offsets")
+  public GetTopicPartitionOffsetResponse getOffsets(
+      @PathParam("topic") String topic,
+      @PathParam("partition") int partition,
+      @QueryParam("timestamp") @Nullable Instant timestamp
+  ) throws Exception {
+    checkTopicExists(topic);
+    checkPartitionExists(topic, partition);
+
+    if (timestamp == null) {
+      return GetTopicPartitionOffsetResponse.createSummary(
+          getBeginningOffset(topic, partition), getEndOffset(topic, partition));
+    }
+
+    return getOffsetForTime(topic, partition, timestamp)
+        .map(
+            offsetTimestamp ->
+                GetTopicPartitionOffsetResponse.createForTime(
+                    offsetTimestamp.getOffset(), offsetTimestamp.getTimestamp()))
+        .orElse(GetTopicPartitionOffsetResponse.empty());
+  }
+
+  /**
+   * Returns the earliest offset in the {@code topic} {@code partition}.
+   */
+  private long getBeginningOffset(String topic, int partition) {
+    try {
+      return retrying(
+          () -> ctx.getKafkaConsumerManager()
+              .getBeginningOffset(consumerGroup, getConsumerInstance(), topic, partition),
```

Comment:
(furthering comments elsewhere) I think the back-and-forth with consumer instance id with the consumer manager is structurally more complex than needed and also there's no scope for other users of consumer manager to re-use these instances.